library/src/main/java/com/parse/twitter/Twitter.java

```diff
@@ -136,27 +143,32 @@ public void authorize(final Context context, final AsyncCallback callback) {
           "Twitter must be initialized with a consumer key and secret before authorization.");
     }
 
-    final OAuthProvider provider = new DefaultOAuthProvider(REQUEST_TOKEN_URL, ACCESS_TOKEN_URL, AUTHORIZE_URL,
-      httpURLConnectionClient);
-    provider.setRequestHeader("User-Agent", USER_AGENT);
-    final OAuthConsumer consumer = new DefaultOAuthConsumer(getConsumerKey(), getConsumerSecret());
-
     final ProgressDialog progress = new ProgressDialog(context);
     progress.setMessage("Loading...");
-    AsyncTask<Void, Void, String> task = new AsyncTask<Void, Void, String>() {
-      private Throwable error;
 
+    AsyncTask<Void, Void, HttpURLConnectionClient> createClient = new AsyncTask<Void, Void, HttpURLConnectionClient>() {
```

Comment:
This outer `AsyncTask` doesn't seem necessary, since the `OAuthProvider`/`Consumer` doesn't need to be created on the UI thread and we could just create the client, provider, and consumer in the background method of the original `AsyncTask`.